pkg/mutate/manage_pod_mutation.go

```diff
@@ -0,0 +1,132 @@
+				corev1.NodeSelectorRequirement{
+					Key:      liqoconst.TypeLabel,
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{liqoconst.TypeNode},
+				})
+		}
+	case nsoff.Spec.PodOffloadingStrategy == offv1alpha1.LocalAndRemotePodOffloadingStrategyType:
+		nodeSelector = nsoff.Spec.ClusterSelector
+		// A new NodeSelectorTerm terms that allows scheduling the pod also on local nodes is added.
+		newNodeSelectorTerm := corev1.NodeSelectorTerm{
+			MatchExpressions: append([]corev1.NodeSelectorRequirement{}, corev1.NodeSelectorRequirement{
+				Key:      liqoconst.TypeLabel,
+				Operator: corev1.NodeSelectorOpNotIn,
+				Values:   []string{liqoconst.TypeNode},
+			}),
+		}
+		nodeSelector.NodeSelectorTerms = append(nodeSelector.NodeSelectorTerms, newNodeSelectorTerm)
+	case nsoff.Spec.PodOffloadingStrategy == offv1alpha1.LocalPodOffloadingStrategyType:
+		// In local strategy there is no necessity to impose a particular NodeSelector.
+		nodeSelector = corev1.NodeSelector{NodeSelectorTerms: []corev1.NodeSelectorTerm{}}
+	}
+	return nodeSelector
+}
+
+// fillPodWithImposedNodeSelector gets the previously computed NodeSelector imposed by the PodOffloadingStrategy and
+// merges it with the Pod NodeSelector if it is already present. It simply adds it to the Pod if there is merge to do.
+func fillPodWithImposedNodeSelector(imposedNodeSelector *corev1.NodeSelector, pod *corev1.Pod) {
+	// To preserve the Pod Affinity content, it is necessary to add the imposedNodeSelector according to what
+	// is already present in the Pod Affinity.
+	switch {
+	case pod.Spec.Affinity == nil:
+		pod.Spec.Affinity = &corev1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{},
+			},
+		}
+	case pod.Spec.Affinity.NodeAffinity == nil:
+		pod.Spec.Affinity.NodeAffinity = &corev1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{},
+		}
+	case pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil:
+		pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &corev1.NodeSelector{}
+	case len(pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms) != 0:
+		oldNodeSelector := *pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+		tmpSelector := corev1.NodeSelector{NodeSelectorTerms: []corev1.NodeSelectorTerm{}}
+		for i := range oldNodeSelector.NodeSelectorTerms {
+			for j := range imposedNodeSelector.NodeSelectorTerms {
+				newMatchExpression := append(imposedNodeSelector.NodeSelectorTerms[j].MatchExpressions, oldNodeSelector.NodeSelectorTerms[i].MatchExpressions...)
+				tmpSelector.NodeSelectorTerms = append(tmpSelector.NodeSelectorTerms, corev1.NodeSelectorTerm{
+					MatchExpressions: newMatchExpression,
+				})
+			}
+		}
```

Comment:
IMHO, this logic deserves to be extracted to a separate function, for the sake of clarity and to simplify testing. More comments should also be added.

Reply:
you are right, I have created a new function